Doc/library/functools.rst

```diff
@@ -430,6 +430,17 @@ The :mod:`functools` module defines the following functions:
      ...     for i, elem in enumerate(arg):
      ...         print(i, elem)
 
+   .. note::
+      The :mod:`typing` module provides a number of aliases to
+      builtin classes, as well as classes found elsewhere in the standard
+      library. All have been deprecated since Python 3.9 (see :pep:`585`).
+      Annotating the first argument of a ``@singledispatch`` implementation
+      with one of these aliases is not supported, and will lead to a
+      :exc:`TypeError` being raised. Use :pep:`585`-style syntax instead:
+      ``list[int]`` rather than ``typing.List[int]``,
```

Comment:
The examples might be read to imply that this only applies to usage with square bracket notation. It's probably clearer to say, don't use typing.List, and then it follows that typing.List[int] will also not work.